### PR TITLE
#90-adjust-crud-functions

### PR DESCRIPTION
**Pull request by MWalve**
This is far from over and has a few errors. but stuck on how to go on from here.

### CODE REVIEW

File: src/types/message.types.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { Topic } from './topic';
  * The format of the data sent from the server
  */
 export type ClientData = {
-  value: string | number;
+  value: string[] | number[];
```
Comment (Peyton-McKee) [inline src/types/message.types.ts]: Keep these as single values for now

File: src/services/data.services.ts [hunk 2]
```diff
@@ -62,13 +66,25 @@ export default class DataService {
...
-    return await prisma.data.create({
+    const createdData = await prisma.data.create({
       data: {
         dataType: { connect: { name: dataType.name } },
         time: unixTime,
         run: { connect: { id: run.id } },
-        value: serverData.value as number
+        value: {
+          createMany: {
+            data: serverData.value.map((value) => ({
```
Comment (Peyton-McKee) [inline src/services/data.services.ts]: Does tjis work?

File: src/services/data.services.ts [hunk 3]
```diff
@@ -62,13 +66,25 @@ export default class DataService {
...
+              value: value.value as number,
+              unit: value.unit
```
Comment (Peyton-McKee) [inline src/services/data.services.ts]: This shouldnt habe a unit right?

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]

File: src/services/data.services.ts (hunk 3 above)
Comment (MWalve) [reply src/services/data.services.ts]: im pretty sure severdata.ts also has value and units. doesnt seem to break when I remove it though

**Review by Peyton-McKee — APPROVED** [review]
looks good

**Pull request merged**